MH4U Dex/CoreDataController.m

```diff
@@ -0,0 +1,430 @@
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K == %@", @"combinedName", combinedName];
+    [request setPredicate:predicate];
+    
+    NSArray *results = [self.managedObjectContext executeFetchRequest:request error:&error];
+    if (!error) {
+        if (results.count == 0) {
+            // Object not found.
+            return nil;
+        } else if (results.count == 1) {
+            return results.firstObject;
+        } else {
+            NSLog(@"Too many entities found!");
+            return nil;
+        }
+    } else {
+        NSLog(@"Error fetching object!");
+        NSLog(@"%@, %@", error, error.localizedDescription);
+        return nil;
+    }
+}
+
+- (NSManagedObject *)getUniqueEntityWithEntityName:(NSString *)entityName withNameAttribute:(NSString *)name
+{
+    NSError *error = nil;
+    NSFetchRequest *request = [[NSFetchRequest alloc] initWithEntityName:entityName];
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K == %@", @"name", name];
+    [request setPredicate:predicate];
+    
+    NSArray *results = [self.managedObjectContext executeFetchRequest:request error:&error];
+    if (!error) {
+        if (results.count == 0) {
+            // Object not found.
+            return nil;
+        } else if (results.count == 1) {
+            return results.firstObject;
+        } else {
+            NSLog(@"Too many entities found!");
+            return nil;
+        }
+    } else {
+        NSLog(@"Error fetching object!");
+        NSLog(@"%@, %@", error, error.localizedDescription);
+        return nil;
+    }
+}
+
+- (NSManagedObject *)getUniqueEntityWithEntityName:(NSString *)entityName withIDNumber:(NSNumber *)entity_id
+{
+    NSError *error = nil;
+    NSFetchRequest *request = [[NSFetchRequest alloc] initWithEntityName:entityName];
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K == %d", @"id", entity_id.intValue];
+    [request setPredicate:predicate];
+    
+    NSArray *results = [self.managedObjectContext executeFetchRequest:request error:&error];
+    if (!error) {
+        if (results.count == 0) {
+            return nil;
+        } else if (results.count == 1) {
+            return results.firstObject;
+        } else {
+            NSLog(@"Too many entities found!");
+            return nil;
+        }
+    } else {
+        NSLog(@"Error fetching object!");
+        NSLog(@"%@, %@", error, error.localizedDescription);
+        return nil;
+    }
+}
+
+- (NSUInteger)checkCountForEntityWithEntityName:(NSString *)entityName withNameAttribute:(NSString *)name
+{
+    NSError *error = nil;
+    NSFetchRequest *request = [[NSFetchRequest alloc] initWithEntityName:entityName];
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K == %@", @"name", name];
+    [request setPredicate:predicate];
+    
+    NSUInteger count = [self.managedObjectContext countForFetchRequest:request error:&error];
+    if (!error) {
```

Comment:
[The docs for `countForFetchRequest:error:`](https://developer.apple.com/library/ios/documentation/Cocoa/Reference/CoreDataFramework/Classes/NSManagedObjectContext_Class/index.html#//apple_ref/occ/instm/NSManagedObjectContext/countForFetchRequest:error:) say that if an error occurs, `count` will be...?